File: Calcutta Node App/flutter_app/lib/screens/services/service_detail_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';
import '../../constants/colors.dart';
import '../../constants/config.dart';
import '../../providers/auth_provider.dart';
import '../../api/client.dart';

class ServiceDetailScreen extends StatelessWidget {
  const ServiceDetailScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final service = ModalRoute.of(context)?.settings.arguments as Map<String, dynamic>;
    final auth = context.watch<AuthProvider>();
    final features = service['features'] as List<dynamic>? ?? [];
    final price = service['price'] ?? 0;

    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        title: const Text('Service Details', style: TextStyle(color: AppColors.textPrimary)),
        backgroundColor: AppColors.surface,
        iconTheme: const IconThemeData(color: AppColors.textPrimary),
      ),
      body: ListView(
        padding: const EdgeInsets.all(20),
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text('${service['category']}', style: const TextStyle(fontSize: 12, color: AppColors.electricViolet, fontWeight: FontWeight.w600, letterSpacing: 1)),
              Text('₹${price is int ? price.toString().replaceAllMapped(RegExp(r'(\d)(?=(\d{3})+(?!\d))'), (m) => '${m[1]},') : price}',
                  style: const TextStyle(fontSize: 24, fontWeight: FontWeight.w800, color: AppColors.neonCyan)),
            ],
          ),
          const SizedBox(height: 12),
          Text('${service['name']}', style: const TextStyle(fontSize: 24, fontWeight: FontWeight.w800, color: AppColors.textPrimary)),
          const SizedBox(height: 12),
          Text('${service['description']}', style: const TextStyle(fontSize: 14, color: AppColors.textSecondary, height: 1.6)),
          if (features.isNotEmpty) ...[
            const SizedBox(height: 24),
            Container(
              padding: const EdgeInsets.all(20),
              decoration: BoxDecoration(
                color: AppColors.cardBg,
                borderRadius: BorderRadius.circular(16),
                border: Border.all(color: AppColors.border),
              ),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  const Text("What's Included", style: TextStyle(fontSize: 15, fontWeight: FontWeight.w700, color: AppColors.textPrimary)),
                  const SizedBox(height: 14),
                  ...features.map((f) => Padding(
                    padding: const EdgeInsets.only(bottom: 10),
                    child: Row(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        const Text('✓ ', style: TextStyle(fontSize: 14, color: AppColors.neonCyan, fontWeight: FontWeight.w700)),
                        Expanded(child: Text('$f', style: const TextStyle(fontSize: 14, color: AppColors.textSecondary))),
                      ],
                    ),
                  )),
                ],
              ),
            ),
          ],
          const SizedBox(height: 28),
          SizedBox(
            width: double.infinity,
            child: ElevatedButton(
              onPressed: () async {
                if (!auth.isAuthenticated) {
                  Navigator.pushNamed(context, '/login');
                  return;
                }
                try {
                  final token = await ApiClient.getToken();
                  final res = await http.post(
                    Uri.parse('${AppConfig.apiBaseUrl}/dashboard/order'),
                    headers: {'Content-Type': 'application/json', if (token != null) 'Authorization': 'Bearer $token'},
                    body: jsonEncode({'serviceId': service['_id']}),
                  );
                  if (res.statusCode == 200 || res.statusCode == 201) {
                    if (context.mounted) {
                      showDialog(context: context, builder: (c) => AlertDialog(
                        backgroundColor: AppColors.surface,
                        title: const Text('Order Placed!', style: TextStyle(color: AppColors.neonCyan)),
                        content: const Text('Your service has been booked.', style: TextStyle(color: AppColors.textSecondary)),
                        actions: [
                          TextButton(onPressed: () { Navigator.pop(c); Navigator.pushNamed(context, '/orders'); }, child: const Text('Track Order', style: TextStyle(color: AppColors.neonCyan))),
                          TextButton(onPressed: () => Navigator.pop(c), child: const Text('OK', style: TextStyle(color: AppColors.textMuted))),
                        ],
                      ));
                    }
                  } else {
                    if (context.mounted) ScaffoldMessenger.of(context).showSnackBar(const SnackBar(content: Text('Failed to place order')));
                  }
                } catch (e) {
                  if (context.mounted) ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text('$e')));
                }
              },
              style: ElevatedButton.styleFrom(
                backgroundColor: AppColors.neonCyan,
                foregroundColor: Colors.black,
                padding: const EdgeInsets.symmetric(vertical: 16),
                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
              ),
              child: const Text('Book This Service', style: TextStyle(fontSize: 16, fontWeight: FontWeight.w700)),
            ),
          ),
        ],
      ),
    );
  }

}

File: app/src/screens/services/ServiceDetailScreen.js
import React from 'react';
import { View, Text, ScrollView, TouchableOpacity, StyleSheet, Alert } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { useAuth } from '../../context/AuthContext';
import api from '../../api/client';
import { colors } from '../../constants/colors';

export default function ServiceDetailScreen({ route, navigation }) {
  const { service } = route.params;
  const { isAuthenticated } = useAuth();

  async function handleBook() {
    if (!isAuthenticated) {
      Alert.alert('Login Required', 'Please login to book a service', [
        { text: 'Cancel', style: 'cancel' },
        { text: 'Login', onPress: () => navigation.navigate('Login') },
      ]);
      return;
    }
    try {
      await api.post('/dashboard/order', { serviceId: service._id });
      Alert.alert('Order Placed!', 'Your service has been booked successfully.', [
        { text: 'Track Order', onPress: () => navigation.navigate('Orders') },
        { text: 'OK' },
      ]);
    } catch (e) {
      Alert.alert('Error', e.response?.data?.message || 'Failed to place order');
    }
  }

  return (
    <ScrollView style={styles.container}>
      <LinearGradient colors={['rgba(0,229,255,0.06)', 'transparent']} style={styles.grad} />
      <View style={styles.header}>
        <Text style={styles.category}>{service.category}</Text>
        <Text style={styles.price}>₹{service.price?.toLocaleString()}</Text>
      </View>
      <Text style={styles.title}>{service.name}</Text>
      <Text style={styles.desc}>{service.description}</Text>

      {service.features && service.features.length > 0 && (
        <View style={styles.featuresCard}>
          <Text style={styles.featuresTitle}>What's Included</Text>
          {service.features.map((f, i) => (
            <View key={i} style={styles.featureRow}>
              <Text style={styles.checkMark}>✓</Text>
              <Text style={styles.featureText}>{f}</Text>
            </View>
          ))}
        </View>
      )}

      <TouchableOpacity style={styles.bookButton} onPress={handleBook}>
        <Text style={styles.bookText}>Book This Service</Text>
      </TouchableOpacity>

      <View style={{ height: 60 }} />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: colors.background },
  grad: { position: 'absolute', top: 0, left: 0, right: 0, height: 200 },
  header: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', paddingHorizontal: 20, paddingTop: 60, paddingBottom: 8 },
  category: { fontSize: 12, color: colors.electricViolet, fontWeight: '600', textTransform: 'uppercase', letterSpacing: 1 },
  price: { fontSize: 24, fontWeight: '800', color: colors.neonCyan },
  title: { fontSize: 24, fontWeight: '800', color: colors.textPrimary, paddingHorizontal: 20, marginBottom: 12 },
  desc: { fontSize: 14, color: colors.textSecondary, paddingHorizontal: 20, lineHeight: 22, marginBottom: 24 },
  featuresCard: { marginHorizontal: 20, backgroundColor: colors.cardBg, borderRadius: 16, padding: 20, borderWidth: 1, borderColor: colors.border },
  featuresTitle: { fontSize: 15, fontWeight: '700', color: colors.textPrimary, marginBottom: 14 },
  featureRow: { flexDirection: 'row', alignItems: 'center', marginBottom: 10 },
  checkMark: { fontSize: 14, color: colors.neonCyan, fontWeight: '700', marginRight: 10, width: 20 },
  featureText: { fontSize: 14, color: colors.textSecondary, flex: 1 },
  bookButton: { marginHorizontal: 20, marginTop: 28, backgroundColor: colors.neonCyan, borderRadius: 14, padding: 16, alignItems: 'center' },
  bookText: { color: colors.black, fontSize: 16, fontWeight: '700' },
});
